src/QueryBuilder/CommonFilterQueryBuilder.php

```diff
@@ -36,105 +30,32 @@ class TableFilterQueryBuilder
         Operator::ne => 'NOT IN',
     ];
 
-    public const DEFAULT_CAST_SIZE = 16384;
-
-    private Connection $connection;
-    private ?TableInfo $tableInfo;
-    private ColumnConverter $columnConverter;
+    protected Connection $connection;
+    protected ColumnConverter $columnConverter;
 
     public function __construct(
         Connection $connection,
-        ?TableInfo $tableInfo,
         ColumnConverter $columnConverter
     ) {
         $this->connection = $connection;
-        $this->tableInfo = $tableInfo;
         $this->columnConverter = $columnConverter;
     }
 
-    public function buildQueryFromCommand(
-        PreviewTableCommand $options,
-        string $schemaName
-    ): TableFilterQueryBuilderResponse {
-        $this->assertFilterCombination($options);
-
-        $query = new QueryBuilder($this->connection);
-
-        $this->processChangedConditions($options, $query);
-
-        try {
-            if ($options->getFulltextSearch() !== '') {
-                if ($this->tableInfo === null) {
-                    throw new LogicException('tableInfo variable has to be set to use fulltextSearch');
-                }
-
-                $tableInfoColumns = [];
-                /** @var TableInfo\TableColumn $column */
-                foreach ($this->tableInfo->getColumns() as $column) {
-                    // search only in STRING types
-                    if ($this->getBasetype($column->getType()) === BaseType::STRING) {
-                        $tableInfoColumns[] = $column->getName();
-                    }
-                }
-
-                $this->buildFulltextFilters(
-                    $query,
-                    $options->getFulltextSearch(),
-                    $tableInfoColumns,
-                );
-            } else {
-                $this->processWhereFilters($options->getWhereFilters(), $query);
-            }
-
-            $this->processOrderStatement($options->getOrderBy(), $query);
-        } catch (QueryException $e) {
-            throw new TableFilterQueryBuilderException(
-                $e->getMessage(),
-                $e
-            );
-        }
-
-        $this->processSelectStatement($options, $query);
-        $this->processLimitStatement($options, $query);
-        $this->processFromStatement($options, $query, $schemaName);
-
-        $sql = $query->getSQL();
-
-        /** @var string[] $types */
-        $types = $query->getParameterTypes();
-
-        return new TableFilterQueryBuilderResponse(
-            $sql,
-            $query->getParameters(),
-            $types,
-            ProtobufHelper::repeatedStringToArray($options->getColumns()),
-        );
-    }
-
-    private function assertFilterCombination(PreviewTableCommand $options): void
+    protected function processChangedConditions(string $changeSince, string $changeUntil, QueryBuilder $query): void
```

Comment:
nechces tam radeji posilat cele options misto tech scalaru ? https://keboola.atlassian.net/wiki/spaces/TECH/pages/905445396/Writing+new+services

Reply:
bohuzel to nejde, protoze
- preview to ma v TablePreviewCommand
- export to ma v ExportOptions